### PR TITLE
Add Error太郎

### PR DESCRIPTION
**Pull request by morikuni**

### CODE REVIEW

File: errors.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+// Error太郎 represents an error as 太郎.
+type Error太郎 struct {
+	err error
+}
+
+// Error returns a string decorated in 太郎 format.
+// This methods implements error interface.
+func (e Error太郎) Error() string {
+	return fmt.Sprintf("%s太郎", e.err.Error())
```
Comment (mizkei) [inline errors.go:22]:
```suggestion
	return fmt.Sprintf("%s太郎", e.err)
```

https://golang.org/pkg/fmt/#hdr-Printing
>4. If an operand implements the error interface, the Error method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any).

File: errors.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+// Format formats the error according to specified format.
+// The +v format prints the detail of Error太郎, but it works
+// only in iTerm2 for now.
```
Comment (mizkei) [inline errors.go]: How about checking  `TERM_PROGRAM` ?

File: errors.go (hunk 1 above)
Comment (morikuni) [reply errors.go:22]: I know it, but I think this is faster than making it called in fmt package.

File: errors.go (hunk 2 above)
Comment (morikuni) [reply errors.go]: LGTM

File: errors.go (hunk 1 above)
Comment (mizkei) [reply errors.go:22]: Yes.
But, allocation is fewer.

```
> cat main_test.go
package main_test

import (
	"errors"
	"fmt"
	"testing"
)

func BenchmarkFormatError(b *testing.B) {
	err := errors.New("test error")
	for i := 0; i < b.N; i++ {
		s := fmt.Sprintf("err:%s", err)
		_ = s
	}
}

func BenchmarkFormatString(b *testing.B) {
	err := errors.New("test error")
	for i := 0; i < b.N; i++ {
		s := fmt.Sprintf("err:%s", err.Error())
		_ = s
	}
}
> go test -bench . -benchmem
goos: darwin
goarch: amd64
BenchmarkFormatError-8    	10000000	       152 ns/op	      16 B/op	       1 allocs/op
BenchmarkFormatString-8   	20000000	        93.4 ns/op	      32 B/op	       2 allocs/op
PASS
ok  	_/Users/mizkei/go	3.655s
```

I think whichever is good 👍

File: errors.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	"os"
+)
+
+// Wrap wraps given error with Error太郎.
+func Wrap(err error) error {
+	return Error太郎{err}
+}
+
+// Error太郎 represents an error as 太郎.
+type Error太郎 struct {
```
Comment (Code-Hex) [inline errors.go:15]: Good job :+1:

File: errors.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+// This methods implements error interface.
+func (e Error太郎) Error() string {
+	return fmt.Sprintf("%s太郎", e.err.Error())
+}
+
+// Format formats the error according to specified format.
+// The +v format prints the soul of Error太郎 when the program
+// is executed in iTerm.app.
+func (e Error太郎) Format(s fmt.State, verb rune) {
+	isiTerm := os.Getenv("TERM_PROGRAM") == "iTerm.app"
```
Comment (Code-Hex) [inline errors.go:29]: I think better to have in Error太郎 struct.

File: go.mod [hunk 5]
```diff
@@ -0,0 +1,3 @@
+module github.com/vkg/errors
+
+go 1.12
```
Comment (Code-Hex) [inline go.mod:3]: module 太郎 :eyes:

File: errors_test.go [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+	}
+}
+
+func BenchmarkSoul(b *testing.B) {
+	err := Wrap(errors.New("vkg"))
+	os.Setenv("TERM_PROGRAM", "iTerm.app")
+	for i := 0; i < b.N; i++ {
+		_ = fmt.Sprintf("%+v", err)
+	}
+}
```
Comment (morikuni) [inline errors_test.go:30]:
```
$ env GO111MODULE=on go test -v -bench . -benchmem ./...
goos: darwin
goarch: amd64
pkg: github.com/vkg/errors
BenchmarkWithError-12           10000000               182 ns/op              64 B/op          4 allocs/op
BenchmarkWithoutError-12         5000000               380 ns/op              48 B/op          3 allocs/op
BenchmarkSoul-12                  200000              6216 ns/op           12363 B/op          2 allocs/op
PASS
ok      github.com/vkg/errors   5.617s
```

File: errors.go (hunk 4 above)
Comment (morikuni) [reply errors.go:29]: Since Error太郎 is exported, we cannot ensure this value is correctly initialized when a user used the struct directly.

Comment (Code-Hex) [reply errors.go:29]: ~~But we can use the unexported field :+1:~~
I see :pray:

**Review by mizkei — APPROVED** [review]

**Pull request merged**